### ISSUE TITLE
Tooltip accessibility review

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#1078]
Tooltips should be associated back to their referring content via `aria-describedby` attributes which need to point to ID refs in the same shadow tree. This may require that a fallback element is sprouted into the host tree to support having the actual content moved to the `document` to escape clipping.

`role="tooltip"` should also be applied.

**Comment by Westbrook** [comment]
We're testing out a solution for this in https://westbrook-tooltip--spectrum-web-components.netlify.app/components/overlay-trigger.

@jnurthen can you take a look and let me know if the way we're approaching this seems like a viable solution? This is the diff that's making this possible if interesed: https://github.com/adobe/spectrum-web-components/commit/20dc569b7aaf8806316b604d9e789aa2be2a82c3 Working through a timeout failure in the tests, but otherwise I think things are looking like we'd like them to here.

It _looks_ like we're running into some sometimes issues around https://spectrum.adobe.com/page/tooltip/#Immediate-or-delayed-appearance that cause the `aria-describedby` association to not always propagate to the screen reader, even when it's created in the accessibility tree. It appears that this comes up in https://react-spectrum.adobe.com/react-spectrum/Tooltip.html but I may be misunderstanding the situation overall. Should we maybe discuss skipping these delay/timing managers in a keyboard context with the Spectrum team? Is `aria-describedby` secondary enough to not need to worry?

**Comment by Westbrook** [comment]
The current version of https://github.com/adobe/spectrum-web-components/pull/1393 uses a `overlayWillOpenCallback` instead of an `overlayOpenedCallback` which allows for us to associate the Tooltip content with the trigger even before the Tooltip is visually present. It's hard to fully say which of these is better/expected for a screen reader. If such a user is going through an application quickly, without the use of `overlayWillOpenCallback` they might never be exposed to the tooltip content. With the previous approach they'd technically get multiple depths of content, _if_ they work back and forth through the content to get the updated screen reader content.
